src/support/allocators/secure.h

```diff
@@ -40,7 +40,11 @@ struct secure_allocator : public std::allocator<T> {
 
     T* allocate(std::size_t n, const void* hint = 0)
     {
-        return static_cast<T*>(LockedPoolManager::Instance().alloc(sizeof(T) * n));
+        T* allocation = static_cast<T*>(LockedPoolManager::Instance().alloc(sizeof(T) * n));
+        if (!allocation) {
+            throw std::bad_alloc();
```

Comment:
Maybe propagate errno info?

Reply:
I don't think `bad_alloc` have any parameterized ctor:s?

Reply:
Indeed - I guess the other option would be to log it in the `allocate` method.